src/App.tsx

```diff
@@ -14,82 +15,137 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
-
 enum SortType {
   NONE,
   ALPHABET,
   LENGTH,
 }
 
 type ReorderOptions = {
-  sortType: SortType,
-  isReversed: boolean,
+  sortType: SortType;
+  isReversed: boolean;
 };
 
-// Use this function in the render to prepare goods
 export function getReorderedGoods(
   goods: string[],
   { sortType, isReversed }: ReorderOptions,
 ) {
-  // To avoid the original array mutation
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  switch (sortType) {
+    case SortType.ALPHABET:
+      visibleGoods.sort((a, b) => (
+        a.localeCompare(b)));
+      break;
+
+    case SortType.LENGTH:
+      visibleGoods.sort((a, b) => (
+        a.length - b.length));
+      break;
+
+    case SortType.NONE:
+    default:
+      break;
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean;
+  sortType: SortType;
+};
+
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  handleClickSortByAlphabetically = () => {
+    this.setState(state => {
+      return { ...state, sortType: SortType.ALPHABET };
+    });
+  };
+
+  handleClickSortByLength = () => {
+    this.setState(state => {
+      return { ...state, sortType: SortType.LENGTH };
```

Comment:
Why do you need functional setState here ?

Reply:
Good point! i don't need to use func SetState.